### PR TITLE
Fix #1081

### LINKED ISSUE #1081: Default resource attributes and attributes in `OTEL_RESOURCE_ATTRIBUTES` are missing when using `Otlp*LogExporter`
**Describe your environment** 

Enable previous logs

**Steps to reproduce**

Set environment variable `OTEL_RESOURCE_ATTRIBUTES` or not.

**What is the expected behavior?**

Auto add default resource and attributes in `OTEL_RESOURCE_ATTRIBUTES` .

**What is the actual behavior?**

Nothing exported.

### PR DESCRIPTION
**Pull request by owent**
Signed-off-by: owent <admin@owent.net>

Fixes #1081

## Changes

+ Using `opentelemetry::sdk::resource::Resource::Create` to inherit data from default resource and `ResourceDetector`.
+ Add unit test for `OtlpLogRecordable`

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

File: exporters/otlp/src/otlp_log_recordable.cc [hunk 1]
```diff
@@ -17,8 +17,10 @@ namespace otlp
 
 proto::resource::v1::Resource OtlpLogRecordable::ProtoResource() const noexcept
 {
-  // TODO Populate shared resource
-  return private_resource_;
+  proto::resource::v1::Resource proto;
+  OtlpRecordableUtils::PopulateAttribute(
+      &proto, opentelemetry::sdk::resource::Resource::Create(resource_attributes_));
```
Comment (lalitb) [inline exporters/otlp/src/otlp_log_recordable.cc:22]: This is fine for now. But the actual fix should be the improvement in LoggerProvider, to let the resources and instrumentation library passed through it ( similar to what we do with TracerProvider). I will create an issue for the improvements.

File: exporters/otlp/test/otlp_log_recordable_test.cc [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    {
+      EXPECT_EQ(attr.value().string_value(), OPENTELEMETRY_SDK_VERSION);
+      ++found_resource_count;
+    }
+  }
+  EXPECT_EQ(3, found_resource_count);
+}
+
+// Test non-int single types. Int single types are tested using templates (see IntAttributeTest)
+TEST(OtlpLogRecordable, SetSingleAtrribute)
```
Comment (lalitb) [inline exporters/otlp/test/otlp_log_recordable_test.cc:136]:
```suggestion
TEST(OtlpLogRecordable, SetSingleAttribute)
```

File: exporters/otlp/test/otlp_log_recordable_test.cc [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  EXPECT_EQ(rec.log_record().attributes(1).key(), double_key);
+  EXPECT_EQ(rec.log_record().attributes(1).value().double_value(), nostd::get<double>(double_val));
+
+  EXPECT_EQ(rec.log_record().attributes(2).key(), str_key);
+  EXPECT_EQ(rec.log_record().attributes(2).value().string_value(),
+            nostd::get<nostd::string_view>(str_val).data());
+}
+
+// Test non-int array types. Int array types are tested using templates (see IntAttributeTest)
+TEST(OtlpLogRecordable, SetArrayAtrribute)
```
Comment (lalitb) [inline exporters/otlp/test/otlp_log_recordable_test.cc:163]:
```suggestion
TEST(OtlpLogRecordable, SetArrayAttribute)
```

File: exporters/otlp/test/otlp_log_recordable_test.cc [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  {
+    EXPECT_EQ(rec.log_record().attributes(0).value().array_value().values(i).int_value(),
+              int_span[i]);
+  }
+}
+}  // namespace otlp
+}  // namespace exporter
+OPENTELEMETRY_END_NAMESPACE
+
+#endif
```
Comment (lalitb) [inline exporters/otlp/test/otlp_log_recordable_test.cc:237]: Thanks for adding the recordable tests.

**Review by lalitb — APPROVED** [review]

File: exporters/otlp/src/otlp_log_recordable.cc (hunk 1 above)
Comment (owent) [reply exporters/otlp/src/otlp_log_recordable.cc:22]: Agree strongly. I could help to do this if there is any need

File: exporters/otlp/test/otlp_log_recordable_test.cc (hunk 2 above)
Comment (owent) [reply exporters/otlp/test/otlp_log_recordable_test.cc]: Done. I also fix the spell error in `otlp_recordable_test.cc`

File: exporters/otlp/test/otlp_log_recordable_test.cc (hunk 3 above)
Comment (owent) [reply exporters/otlp/test/otlp_log_recordable_test.cc]: Done. I also fix the spell error in `otlp_recordable_test.cc`

File: exporters/otlp/src/otlp_log_recordable.cc (hunk 1 above)
Comment (lalitb) [reply exporters/otlp/src/otlp_log_recordable.cc:22]: Thanks. Have added #1086 if you would like to pick it up.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**